### ISSUE TITLE
Strange Telnet and SSH interface on Ubee DDW36C

### ISSUE THREAD

**Issue by Anonymous941** [jclehner/bcm2-utils#64]
I'm trying to dump my Ubee DDW36C's flash.

I managed to get the username and password via `GatewaySettings.bin`, but when I try to run any command:

```plain
$ bcm2dump -vv -L io.log info '192.168.0.1,technician,R3moteAWGM!g'
bcm2dump v0.9.8-26-gbf8da8b
telnet: received command 250,24
telnet: received command 240,256
telnet: received command 251,1
telnet: received command 251,3

error: telnet: interface auto-detection failed

context:
  ==> 'Ubee Cable Modem'
  ==> (empty)
  ==> 'login:'
  <== ''
  ==> (empty)
  ==> 'password:'
  <== ''
```

Connecting via SSH and Telnet both give me the same results: some strange, limited interface.

```plain
ssh -oKexAlgorithms=+diffie-hellman-group1-sha1 -oHostKeyAlgorithms=+ssh-dss -c aes128-cbc technician@192.168.0.1
technician@192.168.0.1's password: 
Ubee Cable Modem

>?
arpclear            Clear ARP table
arp-storms          ARP Storm configurations
authentication      Authentication Configuration
banner              Banner-text
copy                Copy command
dhcp                DHCP commands
dns                 Assign domain name server
exit                Disconnect
firewall            Enable/Disable firewall
filter-Proxy        Filter Proxy
filter-Cookies      Filter Cookies
filter-JavaApplets  Filter Java Applets
filter-ActiveX      Filter ActiveX
filter-PopupWindows Filter Popup Windows
fragmented-ip       Block fragmented IP packets
ipFlood-Detection   IP Flood Detection
portScan-Detection  Port Scan Detection
host-name           Host name command
interface           Interface commands
nat                 Configure NAT settings
nvram               NVRAM command
oper-mode           Configure the system operation mode
ping                Ping specified IP address
pwd                 Change password
reset               Resetting system
rip                 RIP commands
route               Static route command
show                Show commands
sip-alg             Display SIP_ALG status
set-customer-info   customer-info
snmp-configuration  SNMP Configuration
write               Write configuration to nvram

>show
Incomplete command.

>show nvram    
Invalid : nvram

>nvram get
Invalid : get

>nvram dump
Invalid : dump

>help nvram
Invalid : help

>write
Writing successfully....

>su
Invalid : su

>sudo
Invalid : sudo

>shell
Invalid : shell

>bash
Invalid : bash

>sh
Invalid : sh

>exit
Connection to 192.168.0.1 closed by remote host.
Connection to 192.168.0.1 closed.
```

I'm not sure what to do, or how to dump this router's flash.  TFTP and SNMP both cause the connection to be closed.

---

I tried opening it up, and I've found what appear to be UART wires, but I can't figure out its BAUD rate.  Everything I try just gives me gibberish.

Does anyone have any ideas?

**Comment by Anonymous941** [comment]
I factory reset it and disconnected coax, but I can't use SNMP - it returns `Timeout: No Response from 192.168.0.1.`
@jclehner Any ideas?

**Comment by jclehner** [comment]
>  I can't figure out its BAUD rate. Everything I try just gives me gibberish.

What baud rates did you try? Every Broadcom modem I've worked on used 115200 bps.

**Comment by Anonymous941** [comment]
> What baud rates did you try? Every Broadcom modem I've worked on used 115200 bps.

That's the first one I tried.  I also tried 50, 75, 110, 134, 150, 200, 300, 600, 1200, 2400, 4800, 9600, 19200, 28800, 38400, 57600, 78600, 230400, 460800, 57600 and 921600.  Maybe it's not UART?

I had a breakthrough with that shell though, I figured out that by pressing `?` _after_ typing a command, it tells you the arguments.  `copy` claims to let you copy the "image" (I'm assuming that's the flash) via TFTP, but I tried it and it just said `failed` without attempting to make a connection

But `snmp-configuration` lets me enable SNMP and even set the community string!  The problem is, I have no idea how to enable the standard shell

I'll write a program to scrape the command help, and then once I can disconnect the WiFi (SSH and Telnet only work with coax disconnected) I'll run it and hopefully something will work

**Comment by jclehner** [comment]
> That's the first one I tried. I also tried 50, 75, 110, 134, 150, 200, 300, 600, 1200, 2400, 4800, 9600, 19200, 28800, 38400, 57600, 78600, 230400, 460800, 57600 and 921600. Maybe it's not UART? Or is BPS different from BAUD?

It's the same. The other settings are usually 8 bits of data, no parity, 1 stop bit (aka `8N1`) - maybe this devices uses parity? Or 2 stop bits?

**Comment by Anonymous941** [comment]
Finally got it to stop showing gibberish, but now it's showing me this.

```bash
sudo stty -F /dev/serial0 115200 inlcr
sudo screen /dev/serial0
```

```plain
Error - what Unknown command:  'CM> CM> '

CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> Error - what Unknown command:  'CM> '

Error - what Unknown command:  'Error - what Unknown command:  'CM> ''

CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> CM> 
command:  'CM> '

CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> 
CM> 
CM> CM> 

Errod:  'CM> '

CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> 
CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> 
CM> 
CM> CM> CM> 

Error - what Unknown command:  'CM> CM> '

CM> 
CM> CM> CM> 

Error - what Unknown command:  'CM> CM> '

CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> Error - what Unknown command:  'CM> '

Error - what Unknown command:  'Error - what Unknown command:  'CM> ''

CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> 
CM> 
CM> CM> CM> 

Error - what Unknown command:  'CM> CM> '

CM> 
CM> CM> CM> 

Error - what Unknown command:  'CM> CM> '

CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> Error - what Unknown command:  'CM> '

Error - what Unknown command:  'Error - what Unknown command:  'CM> ''

CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> 
CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> 
CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> CM> 

Error - what Unknown command:  'CM> '

CM> 
CM> 
CM> 
CM> CM> CM> Debian GNU/Linux 12 raspberrypi ttyAMA0

Error - what Unknown command:  'CM> CM> Debian GNU/Linux 12 raspberrypi ttyAMA0'
```

**Comment by Anonymous941** [comment]
Okay, never mind, somehow that issue resolved itself.  Thanks for your help so far, this is progress at least!  Now I'm stuck with it scanning:

```plain
Scanning (pattern) DS Channel at 919750000 Hz...
Scanning (pattern) DS Channel at 913750000 Hz...
Scanning (pattern) DS Channel at 907750000 Hz...
Scanning (pattern) DS Channel at 901750000 Hz...
Scanning (pattern) DS Channel at 895750000 Hz...
Scanning (pattern) DS Channel at 889750000 Hz...
Scanning (pattern) DS Channel at 883750000 Hz...
Scanning (pattern) DS Channel at 877750000 Hz...
Scanning (pattern) DS Channel at 871750000 Hz...
Scanning (pattern) DS Channel at 865750000 Hz...
Scanning (pattern) DS Channel at 859750000 Hz...
Scanning (pattern) DS Channel at 853750000 Hz...
Scanning (pattern) DS Channel at 847750000 Hz...
Scanning (pattern) DS Channel at 841750000 Hz...
Scanning (pattern) DS Channel at 835750000 Hz...
Scanning (pattern) DS Channel at 829750000 Hz...
Scanning (pattern) DS Channel at 823750000 Hz...
Scanning (pattern) DS Channel at 817750000 Hz...
Scanning (pattern) DS Channel at 811750000 Hz...
Scanning (pattern) DS Channel at 805750000 Hz...
Scanning (pattern) DS Channel at 799750000 Hz...
Scanning (pattern) DS Channel at 793750000 Hz...
Scanning (pattern) DS Channel at 787750000 Hz...
Scanning (pattern) DS Channel at 781750000 Hz...
Scanning (pattern) DS Channel at 775750000 Hz...
Scanning (pattern) DS Channel at 769750000 Hz...
Scanning (pattern) DS Channel at 763750000 Hz...
Scanning DS Channel at 267000000 Hz...(from scan list)
Scanning DS Channel at 453000000 Hz...(from scan list)
Scanning DS Channel at 423000000 Hz...(from scan list)
Scanning DS Channel at 405000000 Hz...(from scan list)
Scanning DS Channel at 399000000 Hz...(from scan list)
Scanning DS Channel at 381000000 Hz...(from scan list)
Scanning DS Channel at 363000000 Hz...(from scan list)
Scanning DS Channel at 357000000 Hz...(from scan list)
Scanning DS Channel at 351000000 Hz...(from scan list)
Scanning DS Channel at 345000000 Hz...(from scan list)
Scanning DS Channel at 339000000 Hz...(from scan list)
Scanning DS Channel at 333000000 Hz...(from scan list)
Scanning DS Channel at 327000000 Hz...(from scan list)
Scanning DS Channel at 321000000 Hz...(from scan list)
Scanning DS Channel at 315000000 Hz...(from scan list)
Scanning DS Channel at 279000000 Hz...(from scan list)
Scanning (pattern) DS Channel at 757750000 Hz...
Scanning (pattern) DS Channel at 751750000 Hz...
Scanning (pattern) DS Channel at 745750000 Hz...
Scanning (pattern) DS Channel at 739750000 Hz...
Scanning (pattern) DS Channel at 733750000 Hz...
Scanning (pattern) DS Channel at 727750000 Hz...
Scanning (pattern) DS Channel at 721750000 Hz...
Scanning (pattern) DS Channel at 715750000 Hz...
Scanning (pattern) DS Channel at 709750000 Hz...
Scanning (pattern) DS Channel at 703750000 Hz...
Scanning (pattern) DS Channel at 697750000 Hz...
Scanning (pattern) DS Channel at 691750000 Hz...
Scanning (pattern) DS Channel at 685750000 Hz...
Scanning (pattern) DS Channel at 679750000 Hz...
Scanning (pattern) DS Channel at 673750000 Hz...
Scanning (pattern) DS Channel at 667750000 Hz...
Scanning (pattern) DS Channel at 661750000 Hz...
Scanning (pattern) DS Channel at 655750000 Hz...
Scanning (pattern) DS Channel at 649750000 Hz...
Scanning (pattern) DS Channel at 643750000 Hz...
Scanning (pattern) DS Channel at 637750000 Hz...
Scanning (pattern) DS Channel at 631750000 Hz...
Scanning (pattern) DS Channel at 625750000 Hz...
Scanning (pattern) DS Channel at 619750000 Hz...Scanning (pattern) DS Channel at 613750000 Hz...
Scanning (pattern) DS Channel at 607750000 Hz...
Scanning (pattern) DS Channel at 601750000 Hz...
Scanning (pattern) DS Channel at 595750000 Hz...
Scanning (pattern) DS Channel at 589750000 Hz...
```

**Comment by Anonymous941** [comment]
Wait a minute, never mind!  It's working!  I've got access to a CM shell, finally.  Tomorrow I can try to run `bcm2dump` and hopefully post the flash.  Thanks so much for helping

**Comment by Anonymous941** [comment]
@jclehner [Here's a dump of everything mentioned in the README](https://github.com/Anonymous941/DDW36C), let me know if I can help with writing the profile!

**Comment by Anonymous941** [comment]
Closing as this issue is resolved, continuing on #65
